src/Microsoft.AspNet.WebSockets.Protocol/CommonWebSocket.cs

```diff
@@ -59,6 +60,21 @@ public CommonWebSocket(Stream stream, string subProtocol, TimeSpan keepAliveInte
             }
         }
 
+        // NOTE(anurse): Normally we don't group fields and methods like this but I'd rather do this than have two sections of #ifs.
+#if NET45
+        private readonly static RandomNumberGenerator Random = new RNGCryptoServiceProvider();
+#else
+        private readonly static RandomNumberGenerator Random = PlatformHelper.IsWindows ? new WinCapiRng() : null;
```

Comment:
Just realized that PlatformHelper is useless here because CoreCLR... I'll see what I can figure out.